lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -627,37 +627,41 @@ def do_3d_projection(self, renderer):
             self.update_scalarmappable()
             self._facecolors3d = self._facecolors
 
-        txs, tys, tzs = proj3d._proj_transform_vec(self._vec, renderer.M)
-        xyzlist = [(txs[sl], tys[sl], tzs[sl]) for sl in self._segslices]
+        num_faces, num_face_vertices, _ = self._segments.shape
+        vecs = self._segments.reshape(-1, 3).T
+        _, _, pzs = pvecs = proj3d._proj_transform_vectors(vecs, renderer.M)
+        del vecs  # To avoid typos with pvecs
+        psegments = pvecs.T.reshape(self._segments.shape)
+        assert psegments.shape == (num_faces, num_face_vertices, 3)
```

Comment:
With my suggestion below, this becomes
```suggestion
psegments = proj3d._proj_transform_vectors(self._segments, renderer.M)
```